src/WebJobs.Script.WebHost/BreakingChangeAnalysis/BlobChangeAnalysisStateProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal sealed class BlobChangeAnalysisStateProvider : IChangeAnalysisStateProvider
+    {
+        public const string AnalysisTimestampMetadataName = "LastAnalysisTime";
+
+        private readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
+        private readonly IHostIdProvider _hostIdProvider;
+
+        public BlobChangeAnalysisStateProvider(IConfiguration configuration, IHostIdProvider hostIdProvider, ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryBreakChangeAnalysis);
+        }
+
+        public async Task<ChangeAnalysisState> GetCurrentAsync(CancellationToken cancellationToken)
+        {
+            string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+
+            if (string.IsNullOrEmpty(storageConnectionString) ||
+                !CloudStorageAccount.TryParse(storageConnectionString, out CloudStorageAccount account))
+            {
+                throw new InvalidOperationException($"The {nameof(BlobChangeAnalysisStateProvider)} requires the default storage account '{ConnectionStringNames.Storage}', which is not defined.");
+            }
+
+            string hostId = await _hostIdProvider.GetHostIdAsync(cancellationToken);
+            string analysisBlobPath = $"changeanalysis/{hostId}/sentinel";
+
+            CloudBlobClient blobClient = account.CreateCloudBlobClient();
+            var blobContainer = blobClient.GetContainerReference(ScriptConstants.AzureWebJobsHostsContainerName);
+
+            DateTimeOffset lastAnalysisTime = DateTimeOffset.MinValue;
+
+            var reference = blobContainer.GetBlockBlobReference(analysisBlobPath);
+            bool blobExists = await reference.ExistsAsync(null, null, cancellationToken);
+            if (blobExists)
+            {
+                string lastAnalysisMetadata = reference.Metadata[AnalysisTimestampMetadataName];
```

Comment:
This won't throw if it doesn't exist?

Reply:
It would. Making sure this is handled and adding a test for the scenario (which would have required that blob to be manipulated outside of this process, but totally possible).